File: src/Aether/events/playerEvents.java
package Aether.events;

import Aether.AetherPlayer;
import Aether.Main;
import Aether.entity.FloatingText;
import Aether.entity.Npc;
import cn.nukkit.Player;
import cn.nukkit.event.EventHandler;
import cn.nukkit.event.Listener;
import cn.nukkit.utils.TextFormat;

import java.text.SimpleDateFormat;
import java.util.Date;

public class playerEvents implements Listener {

    private Main getPlugin() {
        return Main.getInstance();
    }

    @EventHandler
    public void playerCreation(cn.nukkit.event.player.PlayerCreationEvent event) {
        event.setPlayerClass(Aether.AetherPlayer.class);
    }

    @EventHandler
    public void join(cn.nukkit.event.player.PlayerJoinEvent event) {
        Player player = event.getPlayer();
        event.setJoinMessage("");
        ((AetherPlayer) player).getConfig().set("last_login", new SimpleDateFormat("MM/dd/yyyy hh:mm a zzz").format(new Date()));

        ((AetherPlayer) player).setupForTeleport();
        new Aether.tasks.sendHub(player, true, TextFormat.YELLOW + "Welcome to", getPlugin().getPrefix()).runTaskLater(getPlugin(), 40);
    }

    @EventHandler
    public void playerTeleport(cn.nukkit.event.player.PlayerTeleportEvent event) {
        Player player = event.getPlayer();
        if (event.getTo().getLevel() != getPlugin().getDefaultLevel()) {
            for (Npc npc : getPlugin().npc.values()) {
                npc.despawnFrom(player);
            }
            for (FloatingText text : getPlugin().texts.values()) {
                text.despawnFrom(player);
            }
        }
    }
}

File: src/Aether/events/playerEvents.php
<?php

declare(strict_types=1);

namespace Aether\events;

use pocketmine\event\Listener;
use pocketmine\Player;
use pocketmine\utils\TextFormat as C;
use Aether\Main;

class playerEvents implements Listener {

  private $plugin;
    
  public function __construct(Main $plugin){
    $this->plugin = $plugin;
  }
    
  public function getPlugin() : Main {
    return $this->plugin;
  }
  
  public function preJoin(\pocketmine\event\player\PlayerPreLoginEvent $e) : void {
  if($this->getPlugin()->hadLoaded !== true){
    $e->getPlayer()->close(Main::PREFIX, C::RED .'Server Has Not Loaded yet.');
    $e->setCancelled(true);
    return;
   }
  }

  public function join(\pocketmine\event\player\PlayerJoinEvent $e) : void {
    $player = $e->getPlayer();
    $e->setJoinMessage('');
    $this->getPlugin()->getPlayer($player)->setup();
  }

  public function quit(\pocketmine\event\player\PlayerQuitEvent $e) : void {
    $player = $e->getPlayer();
    $e->setQuitMessage('');
    $this->getPlugin()->getPlayer($player)->saveData();
  }
}